internal/events/eventstream.go

```diff
@@ -208,9 +208,9 @@ func (a *eventStream) startEventHandlers(resume bool) {
 	go a.batchProcessor()
 	// For a pause/resume, the batch dispatcher goroutine is not terminated, hence no need to start it
 	if !resume {
-		a.updateWG.Add(1) // add a channel for batchDispatcher to inform after it has stopped
 		go a.batchDispatcher()
 	}
+	a.updateWG.Add(1) // add a channel for batchDispatcher to inform after it has stopped
```

Comment:
Note this is a secondary fix I noticed in the code. If the batchDispatcher is already running, it should be still waiting for the batch dispatcher to end on the next update.

Reply:
This would affect pause+resume+update scenario.